### PR TITLE
feat(object attributes):

### PR DESCRIPTION
**Pull request by nogizhopaboroda**
- handle object attributes
  - fix ie bug with shuffled attributes order
  - introduce modulor-element

fixes https://github.com/modulor-js/modulor-html/issues/45

### CODE REVIEW

**Comment by nogizhopaboroda** [comment]
@pankajpatel could you please check this one on your project? should be fully backward-compatible, it's a big change

**Comment by nogizhopaboroda** [comment]
@pankajpatel so, can i merge it?

**Comment by pankajpatel** [comment]
I tried this branch on the local,

it is creating problem with the attribute given as `target='[ref="slider"]'` and error thrown is `Unexpected end of JSON input`

attribute-error

**Comment by pankajpatel** [comment]
And also it started throwing error on `document.querySelector('[id="..."]')`

**Comment by nogizhopaboroda** [comment]
thanks for the report! i'll fix it.
But i didn't understand `And also it started throwing error on document.querySelector('[id="..."]')`
Could you please send the error here?

**Comment by nogizhopaboroda** [comment]
@pankajpatel i fixed this issue. could you please check once again?

**Comment by nogizhopaboroda** [comment]
@pankajpatel so, can i merge it?

**Comment by pankajpatel** [comment]
The object attributes are still not working. E.g.
`${'offer'}=${offer}` gets to `offer="[object Object]"` and shows errors like following:
error

**Comment by nogizhopaboroda** [comment]
Seems like the component doesn't have the `offer` getter/setter, which leads to setting attribute via `element.setAttribute()`, which leads to `[object Object]` value. Is that the case? If not, could you please give more information, otherwise i can't detect where the issue is

**Comment by nogizhopaboroda** [comment]
just for me to understand, in the current version there's no such issue, but in this pull request it exists?

**Comment by nogizhopaboroda** [comment]
the way to understand if it's related to the component initialisation is: put debugger in the component's constructor. if it stops before attribute setting, then it's no related.
but if it's really related to that, then it has nothing to do with the templating engine and you'll definitely need to register the component before you render it

**Comment by nogizhopaboroda** [comment]
@pankajpatel so what is it eventually? the bug of this new version or it was the same behaviour previously?

**Comment by pankajpatel** [comment]
It is with the new version as the code breaks down on the attr values like `container="[id='tooltip-container']"` where there are two `=` signs for a single attribute

error

**Comment by nogizhopaboroda** [comment]
@pankajpatel sorry for the super late answer. fixed the attributes issue, tests:
https://github.com/modulor-js/modulor-html/pull/49/files#diff-d18c12ba43e1cebbb76a12c6bbd42d5fR546

Could you please check whenever you have time?

File: test/html/processing.test.js [hunk 1]
```diff
@@ -311,30 +309,49 @@ describe('replaceDynamicTags', () => {
...
-  ];
-
+  ]
   testSets.forEach((testSet, index) => {
     it(`set #${index}`, () => {
       expect(preprocess(testSet.input)).toBe(testSet.expectation);
     });
   })
 });
+
```
Comment (pankajpatel) [inline test/html/processing.test.js]: comments got left behind

File: test/experiment.test.js [hunk 2]
```diff
@@ -1,40 +1,116 @@
-import { html, render, r, stopNode, Template, containersMap } from '@modulor-js/html';
+//import { html, render, r, stopNode, Template, containersMap } from '@modulor-js/html';
```
Comment (pankajpatel) [inline test/experiment.test.js]: comments

**Comment by pankajpatel** [comment]
@nogizhopaboroda This looks good now and works perfectly fine.

**Comment by nogizhopaboroda** [comment]
thanks! i just need to remove those comments, rebase after the latest changes and push again

**Comment by pankajpatel** [comment]
@nogizhopaboroda If you want, I can resolve the conflicts and push again; what do you think?

**Comment by nogizhopaboroda** [comment]
@pankajpatel sure, thanks!
or if i have time this weekend i'll finally do it and let you know, so that you don't waste time

**Comment by nogizhopaboroda** [comment]
@pankajpatel done, rebased, resolved conflicts

**Comment by pankajpatel** [comment]
Awesome 🎉

Merging it now.

**Comment by nogizhopaboroda** [comment]
i can't believe we finally did it!!

**Pull request merged**